pipeline/matcher_merger/matcher/src/main/scala/weco/pipeline/matcher/MatcherSQSLambda.scala

```diff
@@ -40,14 +41,26 @@ trait MatcherSQSLambda[Config <: ApplicationConfig]
     // each of which contains a set of affected ids
 
     // Next, we must do two things:
-    // 1. notify downstream with all the ids in all the results.
+    // 1. notify downstream with all the MatcherResults.
     // 2. filter out any successful ids from the messagesMap and return the bad messageIds.
     resultsFuture.map {
       results: Iterable[MatcherResult] =>
-        // as a prerequisite for both, flatten sets of ids in MatcherResult into a single set of Strings.
-        val identifiers = results.flatMap(_.allUnderlyingIdentifiers).toSet
-        identifiers.foreach(downstream.notify)
-        findMissingMessages(messagesMap, identifiers)
+        // flatten sets of ids in MatcherResult into a single set of Strings.
+        val initialIdentifiers =
+          results.flatMap(_.allUnderlyingIdentifiers).toSet
+
+        results.foldLeft(initialIdentifiers) {
+          (identifiers, result) =>
+            downstream.notify(result)(MatcherResult.encoder) match {
+              case Success(_) => identifiers
+              // remove from initialIdentifiers list any identifier that was not successfully sent downstream
+              case Failure(_) =>
```

Comment:
I don't think this behaves as expected, as the result of this operation is never returned.